analyses/focal-cn-file-preparation/rna-expression-validation.R

```diff
@@ -0,0 +1,189 @@
+# This script examines RNA-seq expression levels of genes that are called as
+# deletions.
+#
+# Chante Bethell for CCDL 2019
+#
+# #### USAGE
+# This script is intended to be run via the command line from the top directory
+# of the repository as follows:
+#
+# Rscript 'analyses/focal-cn-file-preparation/rna-expression-validation.R'
+
+#### Set Up --------------------------------------------------------------------
+
+# Get `magrittr` pipe
+`%>%` <- dplyr::`%>%`
+
+#### Directories and Files -----------------------------------------------------
+
+# Detect the ".git" folder -- this will in the project root directory.
+# Use this as the root directory to ensure proper sourcing of functions no
+# matter where this is called from
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Set path to results and plots directory
+results_dir <-
+  file.path(root_dir, "analyses", "focal-cn-file-preparation", "results")
+
+plots_dir <-
+  file.path(root_dir, "analyses", "focal-cn-file-preparation", "plots")
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+# Read in data from tsv file (produced in `01-prepare-cn-file.R`)
+cn_df <-
+  readr::read_tsv(file.path(results_dir, "controlfreec_annotated_cn_autosomes.tsv.gz"))
+
+# Read in RNA-seq expression data
+rsem_expression_polyA <-
+  readr::read_rds(file.path(root_dir,
+                            "data",
+                            "pbta-gene-expression-rsem-fpkm.polya.rds"))
+
+rsem_expression_stranded <-
+  readr::read_rds(file.path(
+    root_dir,
+    "data",
+    "pbta-gene-expression-rsem-fpkm.stranded.rds"
+  ))
+
+# Read in metadata
+metadata <-
+  readr::read_tsv(file.path(root_dir, "data", "pbta-histologies.tsv"))
+
+#### Custom Functions ----------------------------------------------------------
+
+# Source script with custom functions
+source(
+  file.path(
+    root_dir,
+    "analyses",
+    "focal-cn-file-preparation",
+    "util",
+    "rna-expression-functions.R"
+  )
+)
+
+#### Join data -----------------------------------------------------------------
+
+# Add metadata to focal CN data.frame
+cn_df_loss_metadata <- cn_df %>%
+  dplyr::filter(status %in% c("loss")) %>%
+  dplyr::inner_join(metadata,
+                    by = c("biospecimen_id" = "Kids_First_Biospecimen_ID")) %>%
+  dplyr::select(
+    gene_symbol,
+    biospecimen_id,
+    status,
+    copy_number,
+    tumor_ploidy,
+    Kids_First_Participant_ID,
+    tumor_descriptor
+  )
+
+cn_df_non_loss_metadata <- cn_df %>%
```

Comment:
We actually don't want this as non-loss. What we really want is all the genes/specimen combos that are not in in the CN data *at all*. So that is going to have to be a different, anti-merge function later.